pkg/cloudprovider/external.go

```diff
@@ -65,8 +67,8 @@ func isExternalFeatureGateEnabled(featureGate *configv1.FeatureGate) (bool, erro
 	disabledFeatureGates := sets.NewString(featureSet.Disabled...)
 	// CustomNoUpgrade will override the deafult enabled feature gates.
 	if featureGate.Spec.FeatureSet == configv1.CustomNoUpgrade && featureGate.Spec.CustomNoUpgrade != nil {
-		enabledFeatureGates = sets.NewString(featureGate.Spec.CustomNoUpgrade.Enabled...)
-		disabledFeatureGates = sets.NewString(featureGate.Spec.CustomNoUpgrade.Disabled...)
+		enabledFeatureGates = sets.NewString(featuregates.FeatureGateNamesToStrings(featureGate.Spec.CustomNoUpgrade.Enabled)...)
```

Comment:
Nit: `sets.NewString` is deprecated in favour of a generic based set implementation, `sets.New`, would probably save you the feature gate to string conversion